units/stats_task_end.go

```diff
@@ -52,51 +55,30 @@ func newTaskEndJob() *collectTaskEndDataJob {
 // NewCollectTaskEndDataJob logs information a task after it
 // completes. It also logs information about the total runtime and instance
 // type, which can be used to measure the cost of running a task.
-func NewCollectTaskEndDataJob(t *task.Task, h *host.Host) amboy.Job {
+func NewCollectTaskEndDataJob(t *task.Task, h *host.Host, p *pod.Pod) amboy.Job {
 	j := newTaskEndJob()
 	j.TaskID = t.Id
 	j.Execution = t.Execution
-	j.HostID = h.Id
+	var id string
+	if h != nil {
+		id = h.Id
+	} else {
+		id = p.ID
+	}
```

Comment:
Rather than storing both host ID and pod ID, would it be preferable to just combine the two and store one runtime env ID? Whether or not it ran in a host or pod can be checked by using the task's execution mode field.